packages/DataTable/stories/examples/Marvel/App.js

```diff
@@ -1,47 +1,105 @@
 import React from "react";
+import SidePanel from "@paprika/sidepanel";
 import DataTable from "../../../src";
 import { fetchMarvelAPI } from "./helpers";
 
 export default function App() {
-  const [offsetLetter] = React.useState(0);
-  const [offset] = React.useState(0);
-  const [, setLetters] = React.useState({});
+  const [offsetLetter, setOffsetLetter] = React.useState(0);
+  const [offset, setOffset] = React.useState(0);
+  const [letters, setLetters] = React.useState({});
   const [data, setData] = React.useState([]);
+  const [row, setRow] = React.useState(null);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isLoading, setIsLoading] = React.useState(false);
 
   React.useEffect(() => {
     async function getData() {
       // will search by "a" then by "b" .... "z" try String.fromCharCode(97 + 0) on devtools
       const letter = String.fromCharCode(97 + offsetLetter);
       const data = await fetchMarvelAPI(letter, offset);
       setLetters(letters => {
-        return { ...letters, [letter]: { total: data.total } };
+        const { count, limit, offset, total } = data.data;
+        return { ...letters, [letter]: { count, limit, offset, total } };
       });
 
       setData(state => {
         return state.concat(data.data.results);
       });
+      setIsLoading(() => false);
     }
+
+    setIsLoading(() => true);
     getData();
   }, [offset, offsetLetter]);
 
   function renderSeries(cell) {
     return cell.series.items.map(item => <span>{item.name}</span>);
   }
 
-  function renderThumbnail(cell) {
-    return `${cell.thumbnail.path}.${cell.thumbnail.extension}`;
+  function renderThumbnail(row) {
+    return `${row.thumbnail.path}.${row.thumbnail.extension}`;
+  }
+
+  function handleExpandedRow(row) {
+    setRow(row);
+    setIsOpen(() => true);
+  }
+
+  function handleSidePanelClose() {
+    setIsOpen(() => false);
+  }
+
+  function handleLoadMore() {
+    const currentLetter = String.fromCharCode(97 + offsetLetter);
+    const { offset: currentOffset, total, count } = letters[currentLetter];
+    if (currentOffset + count <= total) {
+      setOffset(off => off + 1);
+      return;
+    }
+    setOffset(() => 0);
+    setOffsetLetter(off => off + 1);
+  }
+
+  function handleKeyDownArrow(event, row) {
+    setRow(() => row);
+  }
+
+  function handleClickCell(cell, row) {
+    setRow(() => row);
   }
 
   return (
     <>
-      <DataTable data={data} keygen="id">
+      {row && (
```

Comment:
Can this be inside a custom hook later? something like `const row = useCurrentRow()` `const cell = useCurrentCell()`